src/sys/uio.rs

```diff
@@ -108,13 +122,70 @@ pub struct RemoteIoVec {
     pub len: usize,
 }
 
+/// A vector of buffers.
+///
+/// Vectored I/O methods like [`writev`] and [`readv`] use this structure for
+/// both reading and writing.  Each `IoVec` specifies the base address and
+/// length of an area in memory.
+#[deprecated = "`IoVec` is no longer used in the public interface, use `IoSlice` or `IoSliceMut` instead"]
```

Comment:
```suggestion
#[deprecated(
    since = "0.24.0",
    note = "`IoVec` is no longer used in the public interface, use `IoSlice` or `IoSliceMut` instead"
)]
```